Java-Basics.md

```diff
@@ -0,0 +1,442 @@
+Ok, so if `modelS` has all the same properties, I should be able to access `name` or `manufacturerName` right?
+
+```java
+System.out.println(modelS.name);
+```
+Results in :
+
+```asciidoc
+Main.java:13: error: name has private access in Car
+		System.out.println(modelS.name);
+		                         ^
+1 error
+```
+
+Why? Notice we had mentioned `private` before the variable `name` in class `Car`. So since its private, we cant touch it from outside!
+
+There are other kinds of access Modifiers such as public, protected, default etc. [Read more about them here](#TODO).
+
+### Constructors
+
+What's the point then? I should be able to store data in it right?
+
+Thats when we use either Getter / Setter methods or in this case Constructors to initialize a class. Basically every Java Class has a constructor, which is the method which is called first when any object of the class is initialized. Think of it as a bit of starter code.
+
+When you write a Class without any constructor, then Java assumes it has a default constructor :
+
+```java
+
+public class Car {
+	private String name;
+}
+
+Car modelS = new Car();
+```
+
+This initializing with no parameters is a way of calling the default constructor. You can also have a default constructor written yourself this way :
+
+```java
+public class Car {
+	private String name;
+
+	public Car() {
+		name = "Tesla";
+	}
+}
+```
+
+Then, when calling `new Car()`, the variable `name` will get auto-initialized to `"Tesla"`.
+
+Like what Constructors can do? [Dive into it more here](#TODO).
+
+### Methods
+
+`getName()` and `getManufacturerName()` are two "Getter" methods we have used here. Notice unlike JavaScript, we **have** to define the return type of any method we write, otherwise it will fail at compile time. If you do not want a method to return anything, use `void` return type.
+
+```java
+public class Car {
+	private String name;
+
+	public void changeName() {
+		name = "Tesla";
+	}
+}
+```
+As with any other language, methods (or functions, if you are here from JS world) are used often for their modularity and reusability. [Read more about them here](#TODO).
+
+## Inheritance
+
+So great you have successfully created a Car class. But, wait, isnt Tesla cars are supposed to be Electric variants? I want a Electric car class, but it also should have the properties of the original `Car` class.
+
+Solution : Inheritance. Java provides a neat way to "inherit" parent properties :
+
+```java
+public class Car {
+	private String name;
+	private String manufacturerName;
+
+	public Car(String name, String man) {
+		this.name = name;
+		this.manufacturerName = man;
+	}
+  // Getter method
+	public String getName() {
+		return name;
+	}
+  // Getter method
+	public String getManufacturerName() {
+		return manufacturerName;
+	}
+}
+
+public class ElectricCar extends Car {
+	public ElectricCar(String name, String man) {
+		super(name, man);
+  }
+
+	public void charge() {
+	 System.out.println("Charging ...");
```

Comment:
Indent it properly with equal number of spaces, as there are in the earlier code snippets in the above lines.